### PR TITLE
feat(cloud): add shared-instance flag in limit superflag in alpha

### PR DESCRIPTION
**Pull request by NamanJain8**
This PR adds a `shared-instance` flag to `--limit` superflag. 
When set to true (false by default), it will:
- Restrict access to any of the ACL operations like `Login`, add/remove/update user from non-galaxy namespaces.
- Prevent the leaking of environment variables for minio and aws.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7770)

### CODE REVIEW

File: edgraph/access_ee.go [hunk 1]
```diff
@@ -605,14 +609,11 @@ type authPredResult struct {
 }
 
 func authorizePreds(ctx context.Context, userData *userData, preds []string,
-	aclOp *acl.Operation) (*authPredResult, error) {
+	aclOp *acl.Operation) *authPredResult {
```
Comment (NamanJain8) [inline edgraph/access_ee.go:612]: This is just some clean up.

**Review by jarifibrahim — COMMENTED** [review]
The PR looks good to me but we need tests. @NamanJain8 create a docker compose file with the cloud flag and then try to run login/create user, etc on that dgraph cluster.

**Review by manishrjain — CHANGES_REQUESTED** [review]
some early comments. Haven't looked at the rest of the PR yet.

> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7770#-:-MZsndlrDdeQWJ7Y3s4o:bq9t1ev)* status: 0 of 13 files reviewed, 4 unresolved discussions (waiting on @NamanJain8 and @vvbalaji-dgraph)

___
*[dgraph/cmd/alpha/run.go, line 262 at r4](https://reviewable.io/reviews/dgraph-io/dgraph/7770#-MZsmn8rCg9wEF_tDWtO:-MZsmn8rCg9wEF_tDWtP:buzvtnf) ([raw file](https://github.com/dgraph-io/dgraph/blob/0772daef8fd304376277233aafca21e15dde8fa3/dgraph/cmd/alpha/run.go#L262)):*
> ```Go
> 		String())
> 
> 	flag.String("cloud", worker.CloudDefaults, z.NewSuperFlagHelp(worker.CloudDefaults).
> ```

how about put this in the "limit" flag?
___
*[dgraph/cmd/alpha/run.go, line 264 at r4](https://reviewable.io/reviews/dgraph-io/dgraph/7770#-MZsn3On4U7yzFSyryp8:-MZsn3On4U7yzFSyryp9:btq0fz2) ([raw file](https://github.com/dgraph-io/dgraph/blob/0772daef8fd304376277233aafca21e15dde8fa3/dgraph/cmd/alpha/run.go#L264)):*
> ```Go
> 	flag.String("cloud", worker.CloudDefaults, z.NewSuperFlagHelp(worker.CloudDefaults).
> 		Head("Dgraph cloud options").
> 		Flag("disable-non-galaxy",
> ```

probably don't need this. Unless you think the cloud flag would grow to many options. I reckon it's just for "shared-instance" users of the cloud.

So, perhaps `limit=shared-instance=true`, which would then automatically disable a bunch of things.
___
*[testutil/multi_tenancy.go, line 347 at r4](https://reviewable.io/reviews/dgraph-io/dgraph/7770#-MZsnPZA329NVz3AxD0x:-MZsnPZA329NVz3AxD0y:b-1ex4r) ([raw file](https://github.com/dgraph-io/dgraph/blob/0772daef8fd304376277233aafca21e15dde8fa3/testutil/multi_tenancy.go#L347)):*
> ```Go
> 
> func Export(t *testing.T, token *HttpToken, dest, accessKey, secretKey string) *GraphQLResponse {
> 	exportRequest := `mutation export($dst: String!, $f: String!, $access: String!, $secret: String!){
> ```

100 chars.

**Review by jarifibrahim — APPROVED** [review]
Get @ahsanbarkati's review as well.

**Review by NamanJain8 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7770)* status: 0 of 14 files reviewed, 3 unresolved discussions (waiting on @manishrjain and @vvbalaji-dgraph)

___
*[dgraph/cmd/alpha/run.go, line 262 at r4](https://reviewable.io/reviews/dgraph-io/dgraph/7770#-MZsmn8rCg9wEF_tDWtO:-MZvRbxBATirjY0KVq7C:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/0772daef8fd304376277233aafca21e15dde8fa3/dgraph/cmd/alpha/run.go#L262)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

how about put this in the "limit" flag?
</blockquote></details>

Done.
___
*[dgraph/cmd/alpha/run.go, line 264 at r4](https://reviewable.io/reviews/dgraph-io/dgraph/7770#-MZsn3On4U7yzFSyryp8:-MZvRdc8AK53U5hMlOuQ:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/0772daef8fd304376277233aafca21e15dde8fa3/dgraph/cmd/alpha/run.go#L264)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

probably don't need this. Unless you think the cloud flag would grow to many options. I reckon it's just for "shared-instance" users of the cloud.

So, perhaps `limit=shared-instance=true`, which would then automatically disable a bunch of things.
</blockquote></details>

Done.
___
*[testutil/multi_tenancy.go, line 347 at r4](https://reviewable.io/reviews/dgraph-io/dgraph/7770#-MZsnPZA329NVz3AxD0x:-MZvRgZT3yTDGCS6dI6d:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/0772daef8fd304376277233aafca21e15dde8fa3/testutil/multi_tenancy.go#L347)):*
<details><summary><i>Previously, manishrjain (Manish R Jain) wrote…</i></summary><blockquote>

100 chars.
</blockquote></details>

Done.

**Review by ahsanbarkati — APPROVED** [review]

**Pull request merged**